portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,13 +20,46 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.gson.Gson;
+import java.util.ArrayList;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    public class Message{
```

Comment:
Is there a particular reason to put this class as a nested class of DataServlet? Unless with a good reason, I think we usually have a separate file for new classes.

Also, it may be good to reconsider using "Message" as the class name. Such names are usually very overloaded and used by many low-level framework code as well. Try to be more concrete. For example, if it's about comments, using CommentMessage would be better. But it's your call for other options.